### ISSUE TITLE
Can't access other containers by their names (internal Docker DNS)

### ISSUE THREAD

**Issue by melyux** [haugene/docker-transmission-openvpn#2664]
### Is there a pinned issue for this?

- [X] I have read the pinned issues and could not find my issue

### Is there an existing or similar issue/discussion for this?

- [X] I have searched the existing issues
- [X] I have searched the existing discussions

### Is there any comment in the documentation for this?

- [X] I have read the documentation, especially the FAQ and Troubleshooting parts

### Is this related to a provider?

- [X] I have checked the provider repo for issues
- [X] My issue is NOT related to a provider

### Are you using the latest release?

- [X] I am using the latest release

### Have you tried using the dev branch latest?

- [ ] I have tried using dev branch

### Docker run config used

  transmission:
    image: haugene/transmission-openvpn
    container_name: transmission
    cap_add:
      - NET_ADMIN
    environment:
      - LOCAL_NETWORK=10.0.0.0/8,172.16.0.0/12,192.168.0.0/16
      ...
      - ENABLE_UFW=true
    volumes:
      ...
# ports:

### Current Behavior

Can't ping/access other containers by their names (Docker DNS/service discovery). I can still ping them using their internal Docker IPs though.

### Expected Behavior

Should still be able to access local Docker DNS

### How have you tried to solve the problem?

- `exec`d into this container and tried pinging other containers by name (fail)
- `exec`d into other containers and tried pinging other containers by name (success)
- Tried turning on and off `ENABLE_UFW` to no effect

### Log output

```
root@b07160d60f78:/# ping radarr
ping: radarr: Name or service not known
```

### HW/SW Environment

```markdown
- OS: Debian
- Docker: Docker version 24.0.4, build 3713ee1
```

**Comment by edgd1er** [comment]
No log, very little conf, quite difficult to guess what could be the problem.
My guess, dns is set to a new value not going through docker's dns. Some provider push dns setting through their openvpn file.
https://openvpn.net/community-resources/how-to/ : Pushing DHCP options to clients
The OpenVPN server can push DHCP options such as DNS and WINS server addresses to clients
`push "dhcp-option DNS 10.66.0.4"`
As your provider is not mentioned, the guess need confirmation.

what is the content of /etc/resolv.conf ?

**Comment by melyux** [comment]
`/etc/resolv.conf` has this:

```
# Generated by openvpn for interface tun0
nameserver 10.8.0.1
```

**Comment by edgd1er** [comment]
default docker dns is 127.0.0.11.
seems like your dns is pushed through openvpn config, there is a setting for openvpn to ignore pushed dns.
you may check in the downloaded file if the options to push dns is present.

the solution may be leave the dns as it is and create a second network and attach your container through the second network.
or try what is suggested in the documentation: https://haugene.github.io/docker-transmission-openvpn/tips-tricks/

**Comment by pkishino** [comment]
yeah, as above..your provider has pushed a diff dns which overwrites yours.. search for dns in the documentation linked above.. as by default, the dns is pinned, it will use the dns and only the dns set in resolve.conf.. if your provider pushes one then it will use that (all of this you can of course change..)
but this is not a container issue but a user setup, so I'll move this to Q&A
